src/controllers/customer.js

```diff
@@ -1,53 +1,53 @@
-const { Prisma } = require("@prisma/client")
-const prisma = require('../utils/prisma')
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+const customerDomain = require("../domains/customer");
 
 const createCustomer = async (req, res) => {
-    const {
-        name,
-        phone,
-        email
-    } = req.body
+  const { name, phone, email } = req.body;
 
-    if (!name || !phone || !email) {
+  if (!name || !phone || !email) {
+    return res.status(400).json({
+      error: "Missing fields in request body",
+    });
+  }
+
+  try {
+    const customer = await customerDomain.createCustomer(name, phone, email);
+    res.status(201).json({ customer })
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+            return res.status(409).json({ error: "A customer with the provided email already exists" })
+        }
+    }
+  
+    res.status(500).json({ error: e.message })
+  }
+};
+
+const updateCustomer = async (req, res) => {
+    const { name, contact } = req.body
+    const { id } = req.params
+
+    if (!name) {
         return res.status(400).json({
             error: "Missing fields in request body"
         })
     }
 
     try {
-        /**
-         * This will create a Customer AND create a new Contact, then automatically relate them with each other
-         * @tutorial https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-related-record
-         */
-        const createdCustomer = await prisma.customer.create({
-            data: {
-                name,
-                contact: {
-                    create: {
-                        phone,
-                        email
-                    }
-                }
-            },
-            // We add an `include` outside of the `data` object to make sure the new contact is returned in the result
-            // This is like doing RETURNING in SQL
-            include: { 
-                contact: true
-            }
-        })
-
-        res.status(201).json({ customer: createdCustomer })
+        const customer = await customerDomain.updateCustomer(name, contact, id)
```

Comment:
it looks like you've really modified the legacy code that was here into something much simpler